webcompat/static/js/lib/bugform.js

```diff
@@ -142,7 +142,7 @@ function BugForm() {
     if (!location.search.search(/open=1/) && !location.search.search(/url=/)) {
       return;
     }
-    var urlParam = location.search.match(/url=(.+)/);
+    var urlParam = location.search.match(/url=([^&]*)/);
```

Comment:
Ah yeah. `location.search` is def not what we want to match against, because it doesn't even get the hash content for your example URL:

```
> location.search
"?open=1&url=http://www.openstreetmap.org/search?query=paris"
```

As for the extra layers param, let me test after escaping.